types/tx/types.go

```diff
@@ -97,6 +106,17 @@ func (t *Tx) GetSigners() []sdk.AccAddress {
 		}
 	}
 
+	// ensure any specified fee payer is included in the required signers (at the end)
```

Comment:
What will we do here when we have fee grants working?

Reply:
I think fee grants will be enough of a project that we do some more refactoring then. 

My first idea was "Refactor this code and make changes to the various ante handlers.". However, on reflection, the current functionality is actually quite useful with no more and should not be removed. Maybe fee grants could add a separate field to specify a grant. eg.

```proto
message Fee {
  repeated cosmos.base.v1beta1.Coin amount = 1;
  uint64 gas_limit = 2;
  string payer = 3;
  string granter = 4;
}
```

* `payer` and `granter` unset is as now (paid from first signer)
* `payer` set and `granter` unset is like my PR (paid from specified payer, who is added to signers)
* `payer` set and `granter` set means the payer (who signed) will attempt to pay the fees via a fee grant from granter's account.
* `payer` unset and `granter` set is either invalid or says the first signer attempts to pay from the granter's account 

This is just my first reflection. We could add this `granter` field by 0.40 if you wish, but I would make that a separate PR, and not worry about implementing that until the fee grant module (which may be 0.40.x if you add the field now, or maybe 0.41)

Reply:
I guess that could work. It does add the overhead of another field, but it does seem clearer and allow the additional third use case you've specified.